src/main/java/org/broadinstitute/hellbender/engine/filters/CountedReadFilter.java

```diff
@@ -0,0 +1,165 @@
+package org.broadinstitute.hellbender.engine.filters;
+
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+import java.util.Objects;
+import java.util.function.Predicate;
+import java.util.stream.IntStream;
+
+/**
+ * Wrapper/adapter for ReadFilter that counts the number of reads filtered, and provides a filter count summary.
+ *
+ * For filters with complex predicates that are composed of compound and/or operators, provides counts
+ * by level for each component predicate.
+ *
+ */
+public class CountedReadFilter implements ReadFilter {
```

Comment:
This name seems slightly ungrammatical to my ears -- could we rename to `CountingReadFilter` instead?